### DISCUSSION TITLE
upstream prematurely closed connection while reading response header from upstream

### DISCUSSION THREAD

**Discussion by lagden** [uNetworking/uWebSockets.js#835] [Q&A] [answered]
I upgraded uWebSockets.js from v20.10.0 to v.20.15.0 and the errors started.

So, I did the downgrade to v.20.14.0 and the problem was solved.  
Anybody knows how to fix it?

### Log error (nginx)
```log
[error]
2535134#2535134: *37759344 upstream prematurely closed connection while reading response header from upstream,
client: xxx,
server: xxx,
request: "POST /v1/events HTTP/2.0",
upstream: "http://127.0.0.1:30061/events",
host: "xxx"
```

#### Nginx (proxy config)
```nginx
location /v1/ {
  proxy_set_header Host $host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_buffering off;
  proxy_read_timeout 300s;
  proxy_connect_timeout 120s;
  proxy_pass http://127.0.0.1:30061/;

  # Logs
  access_log  /var/log/nginx/app_access.log;
  error_log  /var/log/nginx/app_error.log;
}```

**Comment by uNetworkingAB** [comment]
Can you send a wireshark capture or similar of the actual request sent to uws

**Reply by uNetworkingAB** [reply]
Nobody is going to figure anything out unless you send the actual request sent to uWS. Use wireshark or tcpdump to see the actual request.

**Comment by uNetworkingAB** [comment]
The most likely problem is you're sending headers that aren't hyphen or alphanumeric, or you're not sending http1.1 upstream. But I cannot do anything with this information, it tells me literally nothing.

**Comment by youfoundKim** [comment, marked as answer]
In case anybody else is facing this issue, I solved it by adding `proxy_http_version 1.1;` in the `location` block.

**Reply by uNetworkingAB** [reply]
Yes nginx defaults to http1.0 which is deprecated since 30 something years. We need better 500-responses
